readinessProbe/readiness/server.go

```diff
@@ -0,0 +1,59 @@
+package readiness
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	v1 "github.com/openshift/ocs-osd-deployer/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	listenAddr          string = ":8081"
+	readinessPath       string = "/readyz/"
+	NamespaceEnvVarName string = "NAMESPACE"
+)
+
+func Logic(client client.Client, managedOCSResource types.NamespacedName, w io.Writer) (bool, error) {
+
+	var managedOCS v1.ManagedOCS
+
+	if err := client.Get(context.Background(), managedOCSResource, &managedOCS); err != nil {
+		w.Write([]byte("failed to get managedocs resource\n"))
+		return false, err
+	}
+
+	if managedOCS.Status.Components.StorageCluster.State != v1.ComponentReady {
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+func RunServer(client client.Client, managedOCSResource types.NamespacedName, w io.Writer) error {
+
+	// Readiness probe is defined here.
+	// From k8s documentation:
+	// "Any code greater than or equal to 200 and less than 400 indicates success."
+	// [indicates that the deployment is ready]
+	// "Any other code indicates failure."
+	// [indicates that the deployment is not ready]
+	http.HandleFunc(readinessPath, func(httpw http.ResponseWriter, req *http.Request) {
+		ready, err := Logic(client, managedOCSResource, w)
+
+		if err != nil {
+			w.Write([]byte("error in readiness logic\n"))
```

Comment:
You are already logging the error in the logic function then again here. This just creates an over cluttered log. 

Instead, you can drop the log from `Logic` and just return it as the actual error using `fmt.Errorf()` then you can log here providing the returned error as part of the message